swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -365,6 +379,8 @@ private ContractMessageProcessingResult combineStatuses(
   private ContractMessageProcessingResult updateContractRecord(
       ContractEntity existingContract, ContractEntity entity, String subscriptionId) {
     if (Objects.equals(entity, existingContract)) {
+      syncSubscriptionForContract(existingContract, subscriptionId);
```

Comment:
this is not super necessary but I think it's a guard logic to have here, so every time we receive a contract that didn't change, we make sure that the subscriptions (that might be unsynced) for this contract are fixed.